ambry-tools/src/integration-test/java/com.github.ambry/clustermap/HelixBootstrapUpgradeToolTest.java

```diff
@@ -172,7 +177,8 @@ public void testIncompleteZKHostInfo() throws Exception {
       Utils.writeJsonObjectToFile(testPartitionLayout.getPartitionLayout().toJSONObject(), partitionLayoutPath);
       try {
         HelixBootstrapUpgradeUtil.bootstrapOrUpgrade(hardwareLayoutPath, partitionLayoutPath, zkLayoutPath,
-            CLUSTER_NAME_PREFIX, dcStr, DEFAULT_MAX_PARTITIONS_PER_RESOURCE, false, false, new HelixAdminFactory());
+            CLUSTER_NAME_PREFIX, dcStr, DEFAULT_MAX_PARTITIONS_PER_RESOURCE, false, false, new HelixAdminFactory(),
+            true);
```

Comment:
I think, maybe, we should also check the error count from validating HelixClusterManager. From current tests, I don't see the verification on functionality of validating HelixClusterManager.

Reply:
`HelixBootstrapUpgradeUtil#validateAndClose()` checks for it, are you suggesting something else?

Reply:
I was suggesting getting the error count from `validatingHelixClusterManager` after running the tool. Looks like the validating piece of code in `validateAndClose()` (see follow) is not covered by this test. 
```
if (validatingHelixClusterManager != null) {
        ensureOrThrow(validatingHelixClusterManager.getErrorCount() == 0,
            "Helix cluster manager should not have encountered any errors");
      }
```